src/handlers/records-query.ts

```diff
@@ -210,4 +214,22 @@ export class RecordsQueryHandler implements MethodHandler {
   private static shouldProtocolAuthorizeQuery(recordsQuery: RecordsQuery): boolean {
     return recordsQuery.authorSignaturePayload!.protocolRole !== undefined;
   }
+
+  /**
+   * Does the recordQuery filter explicitly for published records.
+   * Checks that there is either an explicit datePublished filter, or published is explicitly not set to false.
+   */
+  private static filtersForPublishedRecords(recordsQuery: RecordsQuery): boolean {
+    const { filter } = recordsQuery.message.descriptor;
+    return filter.datePublished !== undefined || filter.published !== false;
+  }
+
+  /**
+   * Does the recordQuery filter explicitly for unpublished records.
+   * Checks that published is not explicitly set to true, and datePublished is explicitly undefined.
+   */
+  private static filtersForUnpublishedRecords(recordsQuery: RecordsQuery): boolean {
+    const { filter } = recordsQuery.message.descriptor;
+    return filter.published !== true && filter.datePublished === undefined;
```

Comment:
Sanity: I have similar confusion as above. The method seems to imply if `published` is `undefined` and `datePublished` is also `undefined`, then it is a filter for unpublished records? Maybe a rename to `filterIncludesUnpublishedRecords()` better matches the intent?

Reply:
@thehenrytsai same response as above, without either filters it will return both published and unpublished values.

With one of either published explicitly true or a datePublished filter set, it will only return `published` records.

Maybe I can make that clearer in the comments?

Reply:
Yup, it makes sense and reads well through the rest of the code.